### PR TITLE
[New Rule] Potential Privileged Escalation via SamAccountName Spoofing

### PR DESCRIPTION
**Pull request by Samirbous**
Identifies a suspicious computer account name rename event, this may indicate an attempt to exploit CVE-2021-42278 to elevated privileges from standard domain user to domain admin privileges. CVE-2021-42278 is a security vulnerability that allows potential attackers to impersonate a domain controller using computer account sAMAccountName spoofing.

https://exploit.ph/cve-2021-42287-cve-2021-42278-weaponisation.html
https://cloudbrothers.info/en/exploit-kerberos-samaccountname-spoofing/
https://github.com/cube0x0/noPac

## EQL

```
iam where event.action == "renamed-user-account" and
  /* machine account name renamed to user like account name */
  winlog.event_data.OldTargetUserName : "*$" and not winlog.event_data.NewTargetUserName : "*$"
```

## Match Example

image

### CODE REVIEW

File: rules/windows/privilege_escalation_samaccountname_spoofing_attack.toml [hunk 1]
```diff
@@ -0,0 +1,67 @@
+[metadata]
+creation_date = "2021/12/12"
+maturity = "production"
+updated_date = "2021/12/12"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies a suspicious computer account name rename event, this may indicate an attempt to exploit CVE-2021-42278 to
+elevated privileges from standard domain user to domain admin privileges. CVE-2021-42278 is a security vulnerability
```
Comment (w0rk3r) [inline rules/windows/privilege_escalation_samaccountname_spoofing_attack.toml:10]:
```suggestion
elevate privileges from standard domain user to domain admin privileges. CVE-2021-42278 is a security vulnerability
```

**Review by w0rk3r — APPROVED** [review]
Awesome

File: rules/windows/privilege_escalation_samaccountname_spoofing_attack.toml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies a suspicious computer account name rename event, this may indicate an attempt to exploit CVE-2021-42278 to
+elevate privileges from standard domain user to domain admin privileges. CVE-2021-42278 is a security vulnerability
+that allows potential attackers to impersonate a domain controller via sAMAccountName attribute spoofing.
+"""
+from = "now-9m"
+index = ["winlogbeat-*", "logs-windows.*"]
```
[the remark this answers is not in the document]
Comment (Samirbous) [reply rules/windows/privilege_escalation_samaccountname_spoofing_attack.toml:14]: windows integration (security logs similar to 4624)

**Review by brokensound77 — APPROVED** [review]

File: rules/windows/privilege_escalation_samaccountname_spoofing_attack.toml (hunk 2 above)
[the remark this answers is not in the document]
Comment (w0rk3r) [reply rules/windows/privilege_escalation_samaccountname_spoofing_attack.toml:14]: The system integration also collects security logs, I think it is the official one that does this now tbh

image

**Pull request merged**